sources/light_client.move

```diff
@@ -286,22 +346,40 @@ fun calc_past_median_time(c: &LightClient, lb: &LightBlock): u32 {
     nth_element(&mut timestamps, size / 2)
 }
 
-fun set_light_block(lc: &mut LightClient, lb: LightBlock) {
-    dof::add(lc.client_id_mut(), lb.height(), lb);
+
+
+// update and query data
+public(package) fun add_light_block(lc: &mut LightClient, lb: LightBlock) {
+    let block_hash = lb.header().block_hash();
+    df::add(lc.client_id_mut(), block_hash, lb);
+
 }
 
-public fun get_light_block(lc: &LightClient, height: u64): &LightBlock {
-    dof::borrow(lc.client_id(), height)
+public(package) fun remove_light_block(lc: &mut LightClient, block_hash: vector<u8>) {
+    df::remove<_, LightBlock>(lc.client_id_mut(), block_hash);
 }
 
-fun exist(lc: &LightClient, height: u64): bool {
-    dof::exists_(lc.client_id(), height)
+public fun get_light_block_by_hash(lc: &LightClient, block_hash: vector<u8>): &LightBlock {
+    // TODO: Can we use option type?
+    df::borrow(lc.client_id(), block_hash)
 }
 
-#[test_only]
-public fun add_light_block(lc: &mut LightClient, lb: LightBlock) {
-    if (lb.height() > lc.finalized_height) {
-        lc.finalized_height= lb.height();
-    };
-    set_light_block(lc, lb);
+public fun exist(lc: &LightClient, block_hash: vector<u8>): bool {
+    let exist = df::exists_(lc.client_id(), block_hash);
+    exist
+}
+
+public(package) fun set_block_header_by_height(c: &mut LightClient, height: u64, block_header: BlockHeader) {
```

Comment:
why do we store headers and light_block s separately? a `LightBlock` contains already a header, so we are duplicating.

Reply:
Yes, I also think about this when writing the source code. It would be better to create a map between height and block hash. Can we do this in the next PR?

Reply:
yes, let's do in another PR. I created a task for this: https://github.com/gonative-cc/move-bitcoin-spv/issues/13
 -> please add TODO here with a link to the task.